### PR TITLE
Widgets: fields

### LINKED ISSUE #304: Widgets: add fields list widget into add component window
Add a list widget for viewing and editing fields in the add component window. 

This is where the stuff like links and streamed data would be stored. As well as adding the widget we should add the data entered in it into the component's model.

### PR DESCRIPTION
**Pull request by rerpha**
### Issue

Closes #304 

### Description of work

Adds a fields list widget into the add component window. This currently supports Scalar fields but nothing else, and requires dialogs for other types. I have created tickets for these. 

The field widget should have auto complete functionality when you start typing and should display any fields in the nexus base class description

### Acceptance Criteria 

Added component_fields and field_type which contain the widget and the list of types respectively.

### UI tests

N/A

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: nexus_constructor/nexus_wrapper.py [hunk 1]
```diff
@@ -108,6 +112,34 @@ def add_component(self, component_type, component_name, description, geometry):
...
+            field_widget = fields.itemWidget(fields.item(i))
+            field_type = field_widget.field_type_combo.currentText()
+            field_name = field_widget.field_name_edit.text()
+            dataset_type = field_widget.value_type_combo.currentText()
+
+            if field_type == FieldType.scalar_dataset.value:
+                component_group.create_dataset(
+                    field_name,
+                    dtype=PYTHON_TO_HDF5[dataset_type],
+                    data=int(field_widget.value_line_edit.text())
```
Comment (matthew-d-jones) [inline nexus_constructor/nexus_wrapper.py]: This doesn't look right, the cast should depend on `dataset_type`.

File: nexus_constructor/add_component_window.py [hunk 2]
```diff
@@ -38,6 +41,7 @@ def __init__(self, nexus_wrapper: NexusWrapper):
         self.nx_classes = make_dictionary_of_class_definitions(
             os.path.abspath(os.path.join(os.curdir, "definitions"))
         )
+        self.possible_fields = self.nx_classes["NXpinhole"]
```
Comment (matthew-d-jones) [inline nexus_constructor/add_component_window.py]: What is the purpose of this line?

**Comment by rerpha** [comment]
Only using this as reference, we should do this using wrappers instead rather than doing it all through the nexus wrapper when "add" is clicked.

**Comment by rerpha** [comment]
Closing, new PR is #359

**Pull request closed without merging**